Core/include/Acts/Utilities/TrackHelpers.hpp

```diff
@@ -53,20 +53,24 @@ Result<typename track_proxy_t::ConstTrackStateProxy> findFirstMeasurementState(
 
   // TODO specialize if track is forward linked
 
-  auto result = Result<TrackStateProxy>::failure(
-      TrackExtrapolationError::CompatibleTrackStateNotFound);
+  std::optional<TrackStateProxy> firstMeasurementOpt;
 
   for (const auto &trackState : track.trackStatesReversed()) {
     bool isMeasurement =
         trackState.typeFlags().test(TrackStateFlag::MeasurementFlag);
     bool isOutlier = trackState.typeFlags().test(TrackStateFlag::OutlierFlag);
 
     if (isMeasurement && !isOutlier) {
-      result = trackState;
+      firstMeasurementOpt = trackState;
```

Comment:
```suggestion
      result = Result<TrackStateProxy>::success(trackState);
```
would that work too?

Reply:
This works:
```suggestion
     return Result<TrackStateProxy>::success(TrackStateProxy{trackState})
```